### PR TITLE
Removed a smattering of compiler warnings

### PR DESCRIPTION
**Pull request by grahamboree**
Unused variables, signed/unsigned mismatching, and a global variable.  Also changed test case include/exclude flags to unsigned shorts to match their usage,

### CODE REVIEW

File: RecastDemo/Source/TestCase.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ void TestCase::doTests(dtNavMesh* navmesh, dtNavMeshQuery* navquery)
 		iter->nstraight = 0;
 		
 		dtQueryFilter filter;
-		filter.setIncludeFlags((unsigned short)iter->includeFlags);
+		filter.setIncludeFlags(iter->includeFlags);
```
Comment (hymerman) [inline RecastDemo/Source/TestCase.cpp]: Typo? Seems weird that this line has changed but the one below hasn't.

Comment (grahamboree) [inline RecastDemo/Source/TestCase.cpp]: Nice catch, thanks!

File: DetourCrowd/Source/DetourObstacleAvoidance.cpp [hunk 2]
```diff
@@ -331,7 +331,7 @@ float dtObstacleAvoidanceQuery::processSample(const float* vcand, const float cs
 	// find the threshold hit time to bail out based on the early out penalty
 	// (see how the penalty is calculated below to understnad)
 	float minPen = minPenalty - vpen - vcpen;
-	float tThresold = ((double)m_params.weightToi/(double)minPen - 0.1) * (double)m_params.horizTime;
+	float tThresold = (m_params.weightToi / minPen - 0.1f) * m_params.horizTime;
```
Comment (grahamboree) [inline DetourCrowd/Source/DetourObstacleAvoidance.cpp]: I'm not sure why these were being cast to double before the division. Maybe it was done in an attempt to increase the precision of the division?  Unless there's something I'm missing, this is generally not the case for IEEE 754 floats.

Comment (hymerman) [inline DetourCrowd/Source/DetourObstacleAvoidance.cpp]: Oh my. This comment just made me re-read this (brilliant) article:

https://randomascii.wordpress.com/2012/03/21/intermediate-floating-point-precision/

So... maybe the casts were a good idea, maybe they weren't. Maybe it depends on how the code is compiled, and a hundred other things. I think it's probably best to keep the behaviour the same if we don't know what the impact will be - and just cast the original back to float explicitly.

Comment (grahamboree) [inline DetourCrowd/Source/DetourObstacleAvoidance.cpp]: Nice article!  I definitely learned a bit about FLOPs.

It looks like 99d1e791ab9898af18eb08b1e5a7983c5065143e introduced this.  Maybe @axelrodR can help clarify the intent here?

Comment (axelrodR) [inline DetourCrowd/Source/DetourObstacleAvoidance.cpp]: I am not sure there is any reason to the cast there, but I don't really remember. Rethinking the subject, all the parameter values composing that formula are not very sensitive to mantissa size and have similar exponent:
 - m_params.weightToi, m_params.horizTime: The numeruical values that make sense are in the range of 0.5-5 (2.5 in the demo). 
 - minPen is a sum of the current minPenalty and several normalized difference in velocity (normalized is key here), so it either float max or in the range of 0-20, with large jumps between values due to angular change in sampled velocity.

 

    On Monday, January 11, 2016 8:47 PM, Graham Pentheny <notifications@github.com> wrote:
 

 In DetourCrowd/Source/DetourObstacleAvoidance.cpp:> @@ -331,7 +331,7 @@ float dtObstacleAvoidanceQuery::processSample(const float* vcand, const float cs
>  	// find the threshold hit time to bail out based on the early out penalty
>  	// (see how the penalty is calculated below to understnad)
>  	float minPen = minPenalty - vpen - vcpen;
> -	float tThresold = ((double)m_params.weightToi/(double)minPen - 0.1) * (double)m_params.horizTime;
> +	float tThresold = (m_params.weightToi / minPen - 0.1f) * m_params.horizTime;
Nice article! I definitely learned a bit about FLOPs.It looks like 99d1e79 introduced this. Maybe @axelrodR can help clarify the intent here?—

Comment (hymerman) [inline DetourCrowd/Source/DetourObstacleAvoidance.cpp]: Cool - sounds like it should be fine to make the change then.

**Comment by hymerman** [comment]
I think the rest of this PR is good to go; I'll merge now.

**Pull request merged**